node/coinstacks/common/api/src/registry.ts

```diff
@@ -0,0 +1,135 @@
+import { Logger } from '@shapeshiftoss/logger'
+import { ConnectionHandler } from './websocket'
+
+type FormatAddressFunc = (address: string) => string
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+type BlockHandler<T = any, T2 = any> = (block: T) => Promise<{ txs: T2 }>
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+type TransactionHandler<T = any, T2 = any> = (tx: T) => Promise<{ addresses: Array<string>; tx: T2 }>
+
+export class Registry {
+  private clients: Record<string, Map<string, void>>
+  private addresses: Record<string, Map<string, ConnectionHandler>>
+
+  private formatAddressFunc: FormatAddressFunc = (address: string) => address.toLowerCase()
+  private handleBlock?: BlockHandler
```

Comment:
require in constructor as undefined is currently a noop rendering the registry essentially useless w/o